USAGE.md

```diff
@@ -1,370 +1,707 @@
+You can get a list of ATT&CK versions programmatically using the github API:
 
 ```python
-def get_all_techniques(src):
-    filt = [Filter('type', '=', 'attack-pattern')]
-    return src.query(filt)
-    
-def get_technique_by_name(src, name):
-    filt = [
-        Filter('type', '=', 'attack-pattern'),
-        Filter('name', '=', name)
-    ]
-    return src.query(filt)
+import requests
+import re
 
-def get_techniques_by_content(src, content):
-    techniques = get_all_techniques(src)
-    return [
-        tech for tech in techniques
-        if content.lower() in tech.description.lower()
-    ]
-    
-get_technique_by_name(fs, 'Rundll32')
-get_techniques_by_content(fs, 'rundll32.exe')
+refToTag = re.compile(r"ATT&CK-v(.*)")
+tags = requests.get("https://api.github.com/repos/mitre/cti/git/refs/tags").json()
+versions = list(map(lambda tag: refToTag.search(tag["ref"]).groups()[0] , filter(lambda tag: "ATT&CK-v" in tag["ref"], tags)))
+# versions = ["1.0", "2.0", ...]
 ```
 
-### Get Techniques added since a certain time
-This example shows how you can use the `Filter` API to only get techniques that have been added to the STIX content since a certain time based on the `created` timestamp. This code could be used within a larger function or script to alert when a new technique has been added to the ATT&CK STIX/TAXII content. The type could also be changed (or removed completely) to return results for different objects.
+## Accessing multiple domains simultaneously
+Because ATT&CK is stored in multiple domains (as of this writing, pre-attack, mobile-attack, and enterprise-attack), the above methodologies will only allow you to work
+with a single domain at a time. While oftentimes the hard separation of domains is advantageous, occasionally it is useful to combine
+domains into a single DataStore. Use any of the methods above to acquire the individual datastores, and then use the following approach to combine them into
+a single CompositeDataSource:
 
 ```python
-def get_techniques_since_time(src, timestamp):
-    filt = [
-        Filter('type', '=', 'attack-pattern'),
-        Filter('created', '>', timestamp)
-    ]
-    return src.query(filt)
-    
-get_techniques_since_time(src, "2018-10-01T00:14:20.652Z")
+src = stix2.CompositeDataSource()
+src.add_data_sources([enterprise_attack_src, pre_attack_src, mobile_attack_src])
 ```
 
-### Get any object by ATT&CK ID
-In this example, the STIX 2.0 type must be passed into the function. Here we query for the group with ATT&CK ID `G0016` (*APT29*).
+You can then use this CompositeDataSource just as you would the DataSource for an individual domain.
+
+# Python Recipes
+Below are example python recipes which can be used to work with ATT&CK data. 
+They assume the existence of an object implementing the DataStore API. Any of the methods outlined in the
+[accessing ATT&CK Data in python](#accessing-ATTCK-Data-in-Python) section above should provide an object implementing this API.
```

Comment:
It's a little nit-picky, but I'd capitalize the label to match the section header: Accessing ATT&CK Data in Python